Documentation/json-schema/distro-support-schema.json

```diff
@@ -2,51 +2,127 @@
     "$schema": "http://json-schema.org/draft-04/schema#",
     "type": "object",
     "properties": {
-      "DISTRO": {
+        "$schema": {
+            "type": "string"
+        }
+    },
+    "$defs": {
+        "executable_command": {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "type": "object",
+            "properties": {
+                "runUnderSudo": {
+                    "type": "boolean",
+                    "description": "true if this command should be executed under sudo on linux"
+                },
+                "commandRoot": {
+                    "type": "string",
+                    "description": "The first word in the command, e.g. if apt-get install foo, it would be the string apt."
+                },
+                "commandParts": {
+                    "type": "array",
+                    "description": "The remaining parts of the command in an array, for apt-get install {0} it would be [install, {0}] as strings"
+                }
+            },
+            "additionalProperties": false

```

Comment:
this is a key change: each structure added this directive to it to force the validation to reject any additional properties found in the structures in the distro support file. this means we must exactly describe the structure here in order to validate.

Reply:
This is super useful and safeguards us for future contributor edits. Thank you a ton Chet, this is pretty awesome and I never would have thought of it!